src/session.rs

```diff
@@ -68,12 +73,39 @@ impl SessionWrapper {
         query: &PreparedStatementWrapper,
         params: Vec<&QueryParameterWrapper>,
```

Comment:
I don't feel I understand how we can pass the JS `Array<T` as `Vec<&T>`. Please investigate how it works.

If this works **and behaves correctly**, then I conclude that we can pass the `Array<T>` as `&[T]`, too. In such case let's do that to save allocations.

Reply:
The trait ``FromNapiValue`` required for the value to be used as a napi function argument is implemented only for ``Vec``. For this reason using slices doesn't work.

Reply:
Here is the implementation of this trait for Vec and I see no reason this will work differently then expected:

https://github.com/napi-rs/napi-rs/blob/7815851b1ebc1e55c8af6e5b4d428fcfb2e11e02/crates/napi/src/bindgen_runtime/js_values/array.rs#L248-L270

Reply:
Now that I think of it, it makes perfect sense. After all, JS `Array<T>` is most probably represented in the memory differently than `&[T]`; at least, objects are held there by reference, not by value.

Reply:
But then, if `Array<T>` is physically an array of pointers, then we could convert it to `&[&T]`. If it's true, then I believe adding conversion of `Array<T>` -> `&[&T]` to napi-rs would be an amazing achievement of your project/thesis.

Reply:
I've added issue for that.